Sources/NIOHTTP2/HTTP2ChannelHandler.swift

```diff
@@ -990,9 +990,13 @@ extension NIOHTTP2Handler {
 #if swift(>=5.7)
     /// The type of all `inboundStreamInitializer` callbacks.
     public typealias StreamInitializer = @Sendable (Channel) -> EventLoopFuture<Void>
+    /// The type of `inboundStreamInitializer` callbacks which return non-void results.
+    public typealias StreamInitializerWithOutput<Output> = @Sendable (Channel) -> EventLoopFuture<Output>
```

Comment:
Normally we try to avoid type aliases because they become quite hard to discover. Personally I would prefer if we actually spell this out all the time.

Reply:
In general I agree but the increased code duplication for the Sendable and non-Sendable versions in already duplicate-heavy code I think would be worse in this case.

Reply:
Fair enough!